### PR TITLE
Feat(): Support static compliation on Linux MUSL

### LINKED ISSUE #2: Cross-compiling from MacOS to x86_64-linux-musl
Hello @rust-or/owners ,

Have you of you attempted to create an AWS Lambda using the library?

### PR DESCRIPTION
**Pull request by dandxy89**
Fixes #2

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -4,11 +4,30 @@ use std::path::PathBuf;
...
+
+    let dst = if musl {
+        Config::new("HiGHS")
+            .define("FAST_BUILD", "ON")
+            .define("SHARED", "OFF")
+            .define("CMAKE_MSVC_RUNTIME_LIBRARY", "MultiThreadedDLL")
+            .define(
+                "CMAKE_CXX_STANDARD_LIBRARIES",
+                "-static-libgcc -static-libstdc++",
+            )
```
Comment (lovasoa) [inline build.rs:22]: Can you remove all the duplicated code and just keep the additional define for musl ?

File: .gitmodules [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [submodule "HiGHS"]
 	path = HiGHS
-	url = git@github.com:ERGO-Code/HiGHS.git
+	url = https://github.com/ERGO-Code/HiGHS.git
```
Comment (lovasoa) [inline .gitmodules:3]: why ?

**Review by lovasoa — CHANGES_REQUESTED** [review]
Can you switch this to a draft until it passes all tests ? It looks like this breaks macos compilation.
Also, can you add a test for musl builds in addition to the existing dylib linux, macos, and windows ?

File: .gitmodules (hunk 2 above)
Comment (dandxy89) [reply .gitmodules:3]: Will be removed on next update - our CI pipeline was having trouble downloading for some reason...

**Comment by lovasoa** [comment]
You can leave a pr opened even when there are still things to change :) That makes it more visible to others

**Comment by dandxy89** [comment]
I've realised I can run the gitflow task on my fork so will continue on there. 

To be honest, I'm close to not continuing with it and continue to use Python for our solvers. 

Will give it one good go this week 😒

**Comment by dandxy89** [comment]
Can you try building the master branch again? 

The MacOS failures don't make sense...

**Comment by dandxy89** [comment]
Second question: are you using this library at work?

**Comment by lovasoa** [comment]
yes we are using at [enexflow](https://www.enexflow.com/) :)

**Comment by lovasoa** [comment]
i relaunched the ci, but I don't think the problem comes from it...

**Pull request closed without merging**